### ISSUE TITLE
BigGAN: implement Evonorm normalization layers

### ISSUE THREAD

**Issue by gwern** [tensorfork/tensorfork#8]
The just-published Evonorm normalization layer paper evolves several replacements for BatchNorm, some of which remove the need for batch statistics (Evonorm-s layers).  

The advantage of using an Evonorm-s is that it 

1. simplifies the software engineering (no need to track and store all of the mean/variance data and store for sample time etc)
2. outperforms BN slightly on BigGAN ImageNet's FID scores
3. potentially offers up to 40% throughput increase by removing the need to sync around batch statistics (which uses up a large amount of our TPU pod time currently)

The downside is that our Evonorm-s implementation is new & untested, and the compare_gan code assumes BN in a number of places (such as the EMA saving code) making it not quite a drop-in replacement for the BN+RELU layers.

**Comment by gwern** [comment]
Sayak Paul experimented with implementing EvoNorm S0/B0 for TensorFlow 2 on [CIFAR-10 with the Mini Inception model](https://app.wandb.ai/sayakpaul/EvoNorm-TensorFlow2/reports/EvoNorm-layers-in-TensorFlow-2--Vmlldzo4Mzk3MQ); he found it could completely overfit (ie reach ~0 loss on the CIFAR-10 training dataset while validation loss increased) compared to BN. 

I interpret this as good news - it means that EvoNorm, as implemented by an independent developer for TF2, works, and is not a minor tweak but is a *huge* increase in model power compared to BN on CIFAR-10, and so for our datasets of millions of images, it should be a useful upgrade. It also provides an independent implementation to compare for bugs. (EN is simple enough that it should be hard to get wrong, but...)

**Comment by gwern** [comment]
An official EvoNorm-S0/B0 implementation is available at https://github.com/tensorflow/tpu/commit/95647b535084fe369bdc9c12bace726cb91970b4#diff-16057781e361556b2798753ef5e08b5c

**Comment by gwern** [comment]
shawwn has begun experimenting with Evonorm-s and thinks he has found a flaw in all the public implementations: use of local variance rather than full batch variance. The latter seems to work better. I've opened a bug on the original Tensorflow TPU repo to ask the authors about that & how trustworthy their version is: https://github.com/tensorflow/tpu/issues/871
